villain/compile.rkt

```diff
@@ -508,11 +508,55 @@
              (compile-e e (cons x c))
              (Add rsp 8)
              (Jmp return))]
-       [(Lit l)
-        (seq (Cmp rax (imm->bits l))
+       [(Lit i)
+        (seq
+             (Mov r8 (imm->bits i))
+             (Cmp rax r8)
              (Jne next)
              (compile-e e c)
              (Jmp return))]
+       [(String s)
+        (let ((true (gensym "true"))
+              (false (gensym "false")))
+          (seq
+           (Mov r8 rax)
+           (And r8 ptr-mask)
+           (Cmp r8 type-string)
+           (Jne next)
+           (Push rax)
+           (compile-string s)
+           (Push rax)
+           (compile-string/symbol-eq type-string true false (cons #f (cons #f c)))
```

Comment:
I'm surprised the code for strings and symbols is so similar.  Symbols should be a pointer comparison.  Strings need to traverse the string for structural equality.  (Although it may be useful to do a quick pointer equality test and fall back on the structural one when the pointers are unequal.)